TWLight/applications/views.py

```diff
@@ -767,6 +767,34 @@ class EvaluateApplicationView(NotDeleted, CoordinatorOrSelf, ToURequired, Update
     success_url = reverse_lazy('applications:list')
 
     def form_valid(self, form):
+        app = self.object
+        status = form.cleaned_data['status']
+        if app.partner.authorization_method == Partner.PROXY:
+            total_accounts_available_for_distribution = None
+            if app.specific_stream is not None:
+                if app.specific_stream.accounts_available is not None:
+                    total_apps_approved_or_sent = User.objects.filter(
+                                          editor__applications__partner=app.partner,
+                                          editor__applications__status__in=(Application.APPROVED, Application.SENT),
+                                          editor__applications__specific_stream=app.specific_stream).count()
+                    total_accounts_available = app.specific_stream.accounts_available
+                    total_accounts_available_for_distribution = total_accounts_available - total_apps_approved_or_sent
+            elif app.partner.accounts_available is not None:
+                total_apps_approved_or_sent = User.objects.filter(
+                                      editor__applications__partner=app.partner,
+                                      editor__applications__status__in=(Application.APPROVED, Application.SENT),
+                                      editor__applications__specific_stream=None).count()
+                total_accounts_available_for_distribution = app.partner.accounts_available - total_apps_approved_or_sent
+
+            if int(status) == Application.APPROVED:
+                try:
+                    assert app.partner.status != Partner.WAITLIST and (total_accounts_available_for_distribution > 0 if total_accounts_available_for_distribution else True)
+                except AssertionError:
```

Comment:
Why `assert` and catch the AssertionError, rather than a simple `if`?

Reply:
no particular reason